dll/win32/shell32/dialogs/recycler_prop.cpp

```diff
@@ -325,7 +334,41 @@ RecycleBinDlg(
                     {
                         uResult = GetDlgItemInt(hwndDlg, 14002, &bSuccess, FALSE);
                         if (bSuccess)
-                            pItem->dwMaxCapacity = uResult;
+                        {
+                            /*
+                                The logical drive is listed within the listview control. Obtain the said
+                                logical drive by checking its item selection index and then get the string text.
+                            */
+                            hWndListView = GetDlgItem(hwndDlg, 14002);
+                            nItemSel = ListView_GetNextItem(hWndListView, -1, LVNI_ALL);
+
+                            if (nItemSel == -1)
+                            {
+                                return FALSE;
+                            }
+
+                            ListView_GetItemText(hWndListView, nItemSel, 0, szDrive, _countof(szDrive));
+
+
+                            /* Retrieve information of available disk space */
+                            GetDiskFreeSpaceExW(szDrive, NULL, NULL, &FreeBytesSpace);
```

Comment:
Question: how does the szDrive string you got back looks like?

Reply:
The string returned by `ListView_GetItemText()` is supposed to be like *(C:)* (the string can be modified by removing the parentheses and add a backslash) just as in the first sub-item of the list view control which is the drive letter itself.

I have added an `ERR()` macro after ListView_GetItemText (below the 350 line) to see what does `szDrive` look like.
```C
ListView_GetItemText(hWndListView, nItemSel, 0, szDrive, _countof(szDrive));
ERR("The logical drive is: %S\n", szDrive);
```
However when I compiled the build and tested it in debugging mode, I saw nothing in the console. I don't know what causes it but could it be that when you lose the selection of the previously selected item `ListView_GetNextItem()` will simply throw an error?
```
(win32ss/gdi/ntgdi/freetype.c:2206) Can't find OS/2 table - not TT font?
(win32ss/gdi/ntgdi/freetype.c:2206) Can't find OS/2 table - not TT font?
err:(dll/win32/shell32/dialogs/recycler_prop.cpp:351) The logical drive is:
```

Reply:
See my reply about the LVNI_SELECTED flag above.